File: src/main/java/com/challenge/restservice/data/DataStore.java
package com.challenge.restservice.data;

import java.util.UUID;
import java.util.concurrent.ConcurrentMap;

public class DataStore {

    static DataStore store = new DataStore();

    ConcurrentMap<String, Receipt> data;

    private DataStore() {

    }

    Receipt find(String s) {
        if (s == null) {
            return null;
        }

        return data.get(s);
    }

    boolean store(Receipt r) {
        if (r == null) {
            return false;
        }

        if (r.id == null) {
            String newId = UUID.randomUUID().toString();
            r.id = newId;
        }

        data.put(r.id, r);
        return false;
    }

}

File: src/main/java/com/challenge/restservice/data/DataStore.kt
package com.challenge.restservice.data

import java.util.*
import java.util.concurrent.ConcurrentHashMap
import java.util.concurrent.ConcurrentMap

class DataStore private constructor() {

    companion object {
        var instance = DataStore()
    }

    var data = ConcurrentHashMap<String, Receipt>()

    fun find(s: String): Receipt? {
        return data!![s]
    }

    fun store(receipt: Receipt): Boolean {
        if (receipt.id == null) {
            receipt.id = UUID.randomUUID().toString()
        }
        data!![receipt.id!!] = receipt
        return false
    }

}